WeScan/Edit/EditScanViewController.swift

```diff
@@ -142,22 +144,18 @@ final class EditScanViewController: UIViewController {
             "inputBottomRight": CIVector(cgPoint: cartesianScaledQuad.topRight)
             ])
         
-        let enhancedImage = filteredImage.applyingAdaptiveThreshold()?.withFixedOrientation()
-        
-        var uiImage: UIImage!
+        let croppedImage = UIImage.from(ciImage: filteredImage)
         
-        // Let's try to generate the CGImage from the CIImage before creating a UIImage.
-        if let cgImage = CIContext(options: nil).createCGImage(filteredImage, from: filteredImage.extent) {
-            uiImage = UIImage(cgImage: cgImage)
-        } else {
-            uiImage = UIImage(ciImage: filteredImage, scale: 1.0, orientation: .up)
+        // Enhanced Image
+        let enhancedImage = filteredImage.applyingAdaptiveThreshold()?.withFixedOrientation()
+        var enhancedScan: ImageScannerScan?
+        if let unwrappedEnhancedImage = enhancedImage {
+            enhancedScan = ImageScannerScan(from: unwrappedEnhancedImage)
```

Comment:
Not completely sure, but you could also write this as:

```swift
let enhancedScan = enhancedImage.compactMap { ImageScannerScan(from: $0) }
```

Reply:
Looks like it works with a slight adjustment of using `flatMap` instead! Thanks for the tip 😄